python-avd/pyavd/_eos_designs/structured_config/network_services/route_maps.py

```diff
@@ -91,64 +70,66 @@ def _route_maps_vrf_default(self: AvdStructuredConfigNetworkServicesProtocol) ->
         Also checked under router_bgp_vrfs to figure out if a route-map should be set on EVPN export.
         """
         if not self._vrf_default_evpn:
-            return None
+            return
 
         if not any([self._vrf_default_ipv4_subnets, self._vrf_default_ipv4_static_routes["static_routes"], self.shared_utils.is_wan_router]):
-            return None
-
-        route_maps = strip_empties_from_list(
-            [
-                self._evpn_export_vrf_default_route_map(),
-                self._bgp_underlay_peers_route_map(),
-                self._redistribute_connected_to_bgp_route_map(),
-                self._redistribute_static_to_bgp_route_map(),
-            ],
-        )
+            return
+
+        self._evpn_export_vrf_default_route_map()
+        self._bgp_underlay_peers_route_map()
+        self._redistribute_connected_to_bgp_route_map()
+        self._redistribute_static_to_bgp_route_map()
+
+    def _route_maps_vrf_default_check(self: AvdStructuredConfigNetworkServicesProtocol) -> bool:
+        if not self._vrf_default_evpn:
+            return False
+
+        if any((self._vrf_default_ipv4_subnets, self._vrf_default_ipv4_static_routes["static_routes"], self.shared_utils.is_wan_router)):
+            return True
+
+        if not self.inputs.underlay_filter_redistribute_connected:
+            return False
 
-        return route_maps or None
+        if self.shared_utils.wan_role and self._vrf_default_ipv4_static_routes["redistribute_in_overlay"]:  # noqa: SIM103
+            return True
+        return False
```

Comment:
@ClausHolbechArista ok to move to this?
```suggestion
        return self.shared_utils.wan_role and self._vrf_default_ipv4_static_routes["redistribute_in_overlay"]
```

Reply:
I had the same comment above :)